src/duplicate.coffee

```diff
@@ -1,7 +1,181 @@
+			)
+
+	# Store an api key in volatile memory
+	setAPIKey: (username, value, respond) ->
+		@getObject(username).key = value
+		@updateBrain()
+		respond "@#{username} API key updated"
+
+	# Output in thread the whether the key matches, by hash
+	checkAPIKey: (username, hash, respond) ->
+		if hash is @hash @getObject(username).key
+			respond "@#{username} API key matched"
+			true
+		else
+			respond "@#{username} API key differed"
+			false
+
+	# Store your account's regex, for non-default setups
+	setPattern: (username, value, respond) ->
+		@getObject(username).pattern = value
+		respond "@#{username} pattern updated"
+
+	# Output in thread your accounts regex
+	viewPattern: (username, respond) ->
+		respond "@#{username} " + @getObject(username).pattern
+
+	# Duplicate an api key into persistent memory
+	persistUser: (username, respond) ->
+		@persists.add username
+		@updateBrain()
+		respond "@#{username} details saved"
+
+	# Output in thread the accounts that have set up keys
+	listUsers: (respond) ->
+		usernames = Object.keys(@apiKeys).map((val) -> '@' + val)
+		respond(JSON.stringify(usernames))
+
+	# Remove a user's details from both volatile and persistent memory
+	forgetUser: (username, respond) ->
+		delete @apiKeys[username]
+		@persists.delete username
+		@updateBrain()
+		respond "@#{username} details forgotten"
+
+	# Output in thread the environment variables
+	viewEnvVars: (respond) ->
+		output = {}
+		for own key of process.env
+			if key.match /(key|token|password)/i
+				output["sha1(#{key})"] = @hash process.env[key]
+			else
+				output[key] = process.env[key]
+		respond(JSON.stringify(output))
+
+	# Output in thread some explanatory notes
+	viewHelp: (respond) ->
+		respond """
+			# Chat duplicator
+				This script monitors your messages and duplicates then to paired rooms in a different service using your identity.
+				## A few promises
+					* Your API key can only ever be compared to a hash, even by your own account.
+					* Your API key will, by default, be stored in volatile memory to lock it to codebase/configuration.
+					* The codebase and environment is fully scrutinisable.
+					* You can stop others within your organisition.  Only use to control runaway bots or colleagues.
+					* Your personal details may all be entered via PM.
+					* This command will return a good documentation string.
+					* Access to your details will ping you, even if it's your account that's doing it.
+					* All read operations are done via the reveal and check keywords.
+						* Reveal shall give full details and never be used for keys.
+						* Check shall compare the stored value to a provided hash.
+				## Caveats
+					* Until https://github.com/resin-io/hubot-as-webot/issues/15 is resolved the PM history is a vulnerability.
+					* If your API key is set to persist then the vulnerabilities also include the redis brain and code changes.
+				## Commands
+					* #{@robot.name} set my key [string]- store the key value
```

Comment:
Add a note here to suggest that people do this in a 1-1? (That's by reading of the idea here, from the caveats above).